sinks/influxdb/driver.go

```diff
@@ -188,54 +179,29 @@ func hashUID(s string) uint64 {
 	return h.Sum64()
 }
 
-func (sink *influxdbSink) storeEventNoColumns(event kube_api.Event) (*influxdb.Series, error) {
-	// Append labels to seriesName instead of adding extra columns
-	seriesName := strings.Replace(eventsSeriesName, "/", "_", -1)
-	labels := make(map[string]string)
-	if event.InvolvedObject.Kind == "Pod" {
-		labels[sink_api.LabelPodId.Key] = string(event.InvolvedObject.UID)
-		labels[sink_api.LabelPodName.Key] = event.InvolvedObject.Name
+func (sink *influxdbSink) StoreTimeseries(timeseries []sink_api.Timeseries) error {
+	if err := sink.createDatabase(); err != nil {
+		return err
 	}
-	labels[sink_api.LabelHostname.Key] = event.Source.Host
-	seriesName = fmt.Sprintf("%s_%s", util.LabelsToString(labels, "_"), seriesName)
-
-	columns := []string{}
-	columns = append(columns, "time")            // Column 0
-	columns = append(columns, "value")           // Column 1
-	columns = append(columns, "sequence_number") // Column 2
-
-	value, err := getEventValue(&event)
-	if err != nil {
-		return nil, err
-	}
-
-	// There's only one point per series for no columns
-	points := make([][]interface{}, 1)
-	points[0] = make([]interface{}, len(columns))
-	points[0][0] = event.LastTimestamp.Time.Round(time.Millisecond).Unix() // Column 0 - time
-	points[0][1] = sink.seqNum.Get(eventsSeriesName)                       // Column 1 - sequence_number
-	points[0][2] = value                                                   // Column 2 - value
-	return &influxdb.Series{
-		Name:    seriesName,
-		Columns: eventColumns,
-		Points:  points,
-	}, nil
-
-}
-
-func (self *influxdbSink) StoreTimeseries(timeseries []sink_api.Timeseries) error {
-	dataPoints := []*influxdb.Series{}
+	dataPoints := make(influxdb.Point, 0, len(timeseries))
 	for index := range timeseries {
-		dataPoints = append(dataPoints, self.metricToSeries(&timeseries[index]))
+		dataPoints = append(dataPoints, sink.metricToPoint(&timeseries[index]))
+	}
+	bp := influxdb.BatchPoints{
+		Points:          dataPoints,
+		Database:        sink.c.dbName,
+		RetentionPolicy: "default",
 	}
-	// TODO: Group all datapoints belonging to a metric into a single series.
 	// TODO: Record the average time taken to flush data.
-	err := self.client.WriteSeriesWithTimePrecision(dataPoints, influxdb.Second)
-	if err != nil {
-		glog.Errorf("failed to write stats to influxDB - %s", err)
-		self.recordWriteFailure()
+	if _, err = sink.client.Write(bp); err != nil {
+		if strings.Contains(err.Error(), dbNotFoundError) {
+			sink.dbExists = false
+		}
+		glog.Errorf("failed to write stats to influxDB - %v", err)
+		sink.recordWriteFailure()
+	} else {
```

Comment:
Like the other `err` comment, it would be more idiomatic to not have an else and do `return nil` at the end.